File: NendMoPubCustomEventSample/NendMoPubCustomEventSample/InterstitialViewController.m
//
//  InterstitialViewController.m
//  NendMoPubCustomEventSample
//
//  Copyright © 2018年 FAN Communications. All rights reserved.
//

#import "InterstitialViewController.h"
#import "MPInterstitialAdController.h"
#import "NendInstanceMediationSettings.h"

static NSString *const kAdUnitId = @"your_ad_unit_id";

@interface InterstitialViewController () <MPInterstitialAdControllerDelegate>
@property (nonatomic) MPInterstitialAdController *interstitial;
@end

@implementation InterstitialViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)dealloc {
    if (self.interstitial) {
        [MPInterstitialAdController removeSharedInterstitialAdController:self.interstitial];
        self.interstitial = nil;
    }
}

- (IBAction)load:(id)sender {
    
    self.interstitial = [MPInterstitialAdController interstitialAdControllerForAdUnitId:kAdUnitId];
    self.interstitial.delegate = self;
    [self.interstitial loadAd];
}

- (IBAction)show:(id)sender {
    if (self.interstitial.ready) {
        [self.interstitial showFromViewController:self];
    }
}

#pragma mark - MPInterstitialAdControllerDelegate
- (void)interstitialDidLoadAd:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s: %@", __FUNCTION__, interstitial);
}

- (void)interstitialDidFailToLoadAd:(MPInterstitialAdController *)interstitial withError:(NSError *)error {
    NSLog(@"%s: %@, %@", __FUNCTION__, interstitial, error.description);
}

- (void)interstitialWillPresent:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s: %@", __FUNCTION__, interstitial);
}

- (void)interstitialDidPresent:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s: %@", __FUNCTION__, interstitial);
}

- (void)interstitialWillDismiss:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s: %@", __FUNCTION__, interstitial);
}

- (void)interstitialDidDismiss:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s: %@", __FUNCTION__, interstitial);
}

- (void)interstitialDidExpire:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s: %@", __FUNCTION__, interstitial);
}

- (void)interstitialDidReceiveTapEvent:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s: %@", __FUNCTION__, interstitial);
}
@end

File: NendMoPubCustomEventSample/NendMoPubCustomEventSwiftSample/InterstitialViewController.swift
//
//  InterstitialViewController.swift
//  NendMoPubCustomEventSwiftSample
//
//  Copyright © 2018年 FAN Communications. All rights reserved.
//

import UIKit

class InterstitialViewController: UIViewController {

    let adUnitId = "your_ad_unit_id"
    var interstitial: MPInterstitialAdController! = nil
    
    override func viewDidLoad() {
        super.viewDidLoad()

        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func loadAd(_ sender: Any) {
        self.interstitial = MPInterstitialAdController.init(forAdUnitId: self.adUnitId)
        self.interstitial.delegate = self
        self.interstitial.loadAd()
    }
    
    @IBAction func showAd(_ sender: Any) {
        if self.interstitial.ready {
            self.interstitial.show(from: self)
        }
    }
}

extension InterstitialViewController: MPInterstitialAdControllerDelegate {
    
    func interstitialDidLoadAd(_ interstitial: MPInterstitialAdController!) {
        print("\(#function) \(String(describing: interstitial))")
    }
    
    func interstitialDidFail(toLoadAd interstitial: MPInterstitialAdController!, withError error: Error!) {
        print("\(#function) \(String(describing: interstitial)), error: \(String(describing: error))")
    }

    func interstitialWillPresent(_ interstitial: MPInterstitialAdController!) {
        print("\(#function) \(String(describing: interstitial))")
    }
    
    func interstitialDidPresent(_ interstitial: MPInterstitialAdController!) {
        print("\(#function) \(String(describing: interstitial))")
    }
    
    func interstitialWillDismiss(_ interstitial: MPInterstitialAdController!) {
        print("\(#function) \(String(describing: interstitial))")
    }
    
    func interstitialDidDismiss(_ interstitial: MPInterstitialAdController!) {
        print("\(#function) \(String(describing: interstitial))")
    }
    func interstitialDidReceiveTapEvent(_ interstitial: MPInterstitialAdController!) {
        print("\(#function) \(String(describing: interstitial))")
    }
}
